python/ray/train/_internal/worker_group.py

```diff
@@ -387,13 +387,9 @@ def sort_workers_by_ip_and_gpu_id(self, _first_ip: Optional[str] = None):
 
         Args:
             _first_ip: The first IP to group by.
-                Hack to avoid OOMs.
-                This is just a temporary solution for Train loading entire checkpoints
-                into memory by ensuring that the rank 0 worker is on the same node as
-                trainable, thus allowing for lazy checkpoint transfer to be used.
-                See https://github.com/ray-project/ray/issues/33073
-                for more context.
-                TODO remove this argument.
+                This ensures that rank 0 worker is on the same node as the
```

Comment:
"Set this to the node IP of the trainer coordinator to ensure that the rank 0 worker is on..."